photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/GaussianProcessSearch.scala

```diff
@@ -64,12 +63,14 @@ class GaussianProcessSearch[T](
 
   private var observedPoints: Option[DenseMatrix[Double]] = None
   private var observedEvals: Option[DenseVector[Double]] = None
-  private var bestEval: Double = evaluator.defaultScore
+  private var bestEval: Double = 0.0
```

Comment:
Won't this be an issue if it's a minimization problem?

Reply:
I don't think so. (1) The evaluation value is demeaned. (2) Looks like over all the evaluators the defaults are always 0.

Reply:
Just to make sure, lets test this out with an AUC and RMSE test function.

Reply:
@joshvfleming Will it cause a numerical overflow?

Reply:
@cmjiang Not if we're using it only for comparison.

Reply:
@joshvfleming I am wondering if we do a calculation with `bestEval` at line 91 `val overallBestEval = Math.min(priorBestEval, bestEval - currentMean)` and `currentMean` is negative, then the result is not expected.

Reply:
@joshvfleming Never mind. The `bestEval` will be updated by line 140 `bestEval = Math.min(bestEval, eval)` before executing line 91 `val overallBestEval = Math.min(priorBestEval, bestEval - currentMean)`.